### PR TITLE
Anahit dev

### PR DESCRIPTION
**Pull request by anahitH**
Implemented features and corresponding activities
    User login, SigninActivity
    User signup, SignupActivity
    User logout, from UserActivity option menu
    User personal data retrieving, UserActivity. The content of AttendancesActivity is moved to UserActivity
    Courses retrieving, CoursesActivity, opens from option menu in UserActivity
    Groups retrieving (partial). Shows all groups of a selected course, opens from CoursesActivity.

Minor changes in QRCodeActivity. Modified AsynchTask of qr code retrieval to extent BaseAsynchTask, so it will have loading bar. Moved the part of getting qr image from the token to background thread, so that ui does not hang.

### CODE REVIEW

File: AAT_android/app/src/main/java/com/ase/aat_android/activities/CourseActivity.java [hunk 1]
```diff
@@ -12,29 +12,113 @@
...
+            url = EndpointUtil.solveUrl(url, "course_id", Long.toString(params[0]));
+            ClientResource groupsRetrieveRes = new ClientResource(Method.GET, url);
+            groupsRetrieveRes.setResponseEntityBuffering(true);
+            groupsRetrieveRes.setRequestEntityBuffering(true);
+            groupsRetrieveRes.accept(MediaType.APPLICATION_ALL_JSON);
+
+            ObjectMapper mapper = new ObjectMapper();
+            List<Object> groupObjects = null;
+            try {
+                // Throws exception.
```
Comment (djwessel) [inline AAT_android/app/src/main/java/com/ase/aat_android/activities/CourseActivity.java]: If this is all commented out, do we even need it?

File: AAT_android/app/src/main/java/com/ase/aat_android/activities/CoursesActivity.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+    /**
+     * ATTENTION: This was auto-generated to implement the App Indexing API.
+     * See https://g.co/AppIndexing/AndroidStudio for more information.
+     */
+    public Action getIndexApiAction() {
+        Thing object = new Thing.Builder()
+                .setName("Courses Page") // TODO: Define a title for the content shown.
+                // TODO: Make sure this auto-generated URL is correct.
+                .setUrl(Uri.parse("http://[ENTER-YOUR-URL-HERE]"))
```
Comment (djwessel) [inline AAT_android/app/src/main/java/com/ase/aat_android/activities/CoursesActivity.java]: What is the purpose of this auto generated code? If we are using it, should we change the url?

**Review by djwessel — APPROVED** [review]

File: AAT_android/app/src/main/java/com/ase/aat_android/activities/CoursesActivity.java [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            String url = EndpointsURL.HTTP_ADDRESS+ EndpointsURL.REQUEST_COURSES;
+            ClientResource coursesRetRes = new ClientResource(Method.GET, url);
+            coursesRetRes.setResponseEntityBuffering(true);
+            coursesRetRes.setRequestEntityBuffering(true);
+            coursesRetRes.accept(MediaType.APPLICATION_ALL_JSON);
+
+            ObjectMapper mapper = new ObjectMapper();
+            List<Object> courseObjects = null;
+            try {
+                courseObjects = mapper.readValue(coursesRetRes.get().getText(), new TypeReference<List<com.aat.datastore.Course>>() {
```
Comment (yemramirezca) [inline AAT_android/app/src/main/java/com/ase/aat_android/activities/CoursesActivity.java]: I tried before something like: 
List<Group> courseObjects =  mapper.readValue(coursesRetRes.get().getText(), new TypeReference<List<Object>>() 

In that case you wouldn't need the method ArrayList<Course> retrieveCourses(List<Object> courseObjects).

File: AAT_android/app/src/main/java/com/ase/aat_android/data/Course.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+/**
+ * Wrapper class for datastore Course
+ * Created by anahitik on 02.02.17.
+ */
+
+public class Course implements Serializable{
+    com.aat.datastore.Course course;
```
Comment (yemramirezca) [inline AAT_android/app/src/main/java/com/ase/aat_android/data/Course.java:16]: Why is necessary a new Course class which only contains a Course?

Comment (anahitH) [reply AAT_android/app/src/main/java/com/ase/aat_android/data/Course.java:16]: This class is used when passing Course object from CoursesActivity to CourseActivity.
Before all fields of datastore Course object were put into the extras (as datastore Course is not implementing Serializable, it is not possible to put its object into an extra).
This class provides easier way to Course object.

**Pull request merged**